scenario_gym/metrics/trajectory.py

```diff
@@ -64,3 +67,813 @@ def _step(self, state: State) -> None:
+                pos_non_ego = state.poses[entity][:2]
+                vel_non_ego = state.velocities[entity][:2]
+                box_non_ego = entity.get_bounding_box_geom(state.poses[entity])
+
+                if entity.__class__.__name__ == 'Pedestrian':
+                    distance_tolerance = 1
+                elif entity.__class__.__name__ == 'Vehicle':
+                    distance_tolerance = 0.1
+
+                # Check if ego will collide with other entity. 
+                self.ego_time_to_collision_bool_and_value = self._will_collide_check(pos_ego, vel_ego, box_ego, pos_non_ego, vel_non_ego, box_non_ego, distance_tolerance)
+
+                # If they will collide, then retrieve time-to-collision. 
+                if self.ego_time_to_collision_bool_and_value[0] == False:
+                    continue
+
+                elif entity.__class__.__name__ != 'Pedestrian' and entity.__class__.__name__ != 'Vehicle':
+                    print(entity.__class__.__name__)
+                    continue
+
+                elif self.ego_time_to_collision_bool_and_value[1] < 1:
+                    ego_class = self.ego.__class__.__name__
+                    entity_class = entity.__class__.__name__
+                    entity_delta_v = (self.predefined_masses[ego_class] / (self.predefined_masses[ego_class] + self.predefined_masses[entity_class])) * (np.linalg.norm(vel_ego) - np.linalg.norm(vel_non_ego))
+
+                    # state.t inclusion in the dictionary (line below). To track/timestamp TTC in simulation.
+                    if state.t not in self.delta_v_dict:
+                        self.delta_v_dict[state.t] = []
+                    self.delta_v_dict[state.t].append({"delta_v" : {f"{entity}": entity_delta_v}})
+                else:
+                    continue 
+        
+    def get_state(self):
+        """Return delta_v dictionary."""
+        return self.delta_v_dict
+    
+
+# #TODOASAP - CONFLICT INDEX
+# class ConflictIndex(Metric):
+#     """
+#     """
+#     name = "conflict_index"
+
+
+#---------- Collision Check ----------#
+
+class CollisionCheck(Metric):
+    """Checks if the ego vehicle collided with any object in the scenario"""
+
+    name = "collision_timestamp"
+
+    def _reset(self, state: State) -> None:
+        self.ego = state.scenario.ego
+        self.collision_check_and_timestamp = False
+        self.t = 0.0
+
+    def _step(self, state: State) -> None:
+        if self.collision_check_and_timestamp == False:
+            # print("Collision check is false")
+            if len(state.collisions()[state.scenario.entities[0]]) > 0:
+                self.collision_check_and_timestamp = state.t
+                # print("collision check is true at time", state.t)
+
+    def get_state(self) -> float:
+        return self.collision_check_and_timestamp
+    
+
+#---------- Cleaup and Lagging Metrics Check ----------#
+
+def cleanup_and_lagging_metrics(data):
```

Comment:
This method looks like data formatting that is very specific to your application and not general enough to live in `trajectory.py`. I suggest that we remove it from this PR and then let's see whether we can find a place for it somewhere together with the rest of your data processing scripts.